ui/app/pages/confirmation/confirmation.js

```diff
@@ -115,7 +115,10 @@ export default function ConfirmationPage() {
   const t = useI18nContext();
   const dispatch = useDispatch();
   const history = useHistory();
-  const pendingConfirmations = useSelector(getUnapprovedConfirmations, isEqual);
+  const pendingConfirmations = useSelector(
```

Comment:
Was this the underlying error? I'm not quite sure I follow :thinking: 

My guess is that the problem occurs when there is a "request permissions" confirmation queued up behind a templated confirmation. I don't yet understand why it only affects the request permissions confirmation though :thinking:

Reply:
`wallet_requestPermissions` uses the approval controller, which creates an entry in the `pendingApprovals` object. I was expecting that type of approval to always be handled by the routing login in the home component, and supersede this new page ever being called. However, if a `wallet_addEtherumChain` is queued up *before* a `wallet_requestPermissions`, the routing logic is skipped because there is already a notification window open and the user is viewing the `addEthereumChain` confirmation. Once they click confirm this page would have attempted to pull the remaining approval out which would be the `wallet_requestPermissions` one.

Reply:
Ahhh I see! That makes sense, thanks